src/view/components/common/iframe-legacy/Iframe-legacy-component.ts

```diff
@@ -27,32 +19,27 @@ export class IframeLegacyComponent extends SiteChangeListener {
     private loadingInProgress: boolean = true;
     private currentId: string;
 
-    constructor(private params$: RouteParams, private routingService: RoutingService,
-                private sanitizer: DomSanitizationService, private element: ElementRef, private siteService: SiteService, private loginService: LoginService) {
+    constructor(private route: ActivatedRoute, private routingService: RoutingService, private siteService: SiteService,
+                private sanitizer: DomSanitizer, private element: ElementRef, private loginService: LoginService) {
         super(siteService);
     }
 
     ngOnInit(): void {
         this.iframeElement = this.element.nativeElement.querySelector('iframe');
 
         if (this.routingService.currentMenu) {
-            this.initComponent( this.routingService.currentMenu );
+            this.initComponent();
         }
 
         this.routingService.menusChange$.subscribe(menus => this.initComponent(menus));
```

Comment:
Why is initComponent called again?, I think that this component don't need to be subscribe to menusChange$ anymore